x/meshsecurity/keeper/querier_test.go

```diff
@@ -39,15 +39,52 @@ func TestQueryVirtualStakingMaxCap(t *testing.T) {
 	}
 	for name, spec := range specs {
 		t.Run(name, func(t *testing.T) {
-			gotRsp, gotErr := NewQuerier(keepers.EncodingConfig.Marshaler, k).VirtualStakingMaxCap(sdk.WrapSDKContext(ctx), &types.QueryVirtualStakingMaxCapRequest{
+			gotRsp, gotErr := NewQuerier(keepers.EncodingConfig.Marshaler, k).VirtualStakingMaxCapLimit(sdk.WrapSDKContext(ctx), &types.QueryVirtualStakingMaxCapLimitRequest{
 				Address: spec.addr,
 			})
 			if spec.expErr {
 				require.Error(t, gotErr)
 				return
 			}
 			require.NoError(t, gotErr)
-			assert.Equal(t, spec.expAmount, gotRsp.Limit)
+			assert.Equal(t, spec.expAmount, gotRsp.Cap)
 		})
 	}
 }
+
+func TestQueryVirtualStakingMaxCapLimits(t *testing.T) {
+	// setup
+	ctx, keepers := CreateDefaultTestInput(t)
+	k := keepers.MeshKeeper
+
+	querier := NewQuerier(keepers.EncodingConfig.Marshaler, k)
+	// when
+	gotRsp, err := querier.VirtualStakingMaxCapLimits(sdk.WrapSDKContext(ctx), &types.QueryVirtualStakingMaxCapLimitsRequest{})
+	// then
+	require.NoError(t, err)
+	assert.Zero(t, len(gotRsp.MaxCapInfos))
+
+	// set max cap for a random contract
+	myContract := sdk.AccAddress(rand.Bytes(32))
+	myAmount := sdk.NewInt64Coin(keepers.StakingKeeper.BondDenom(ctx), 123)
+	err = k.SetMaxCapLimit(ctx, myContract, myAmount)
+	require.NoError(t, err)
+
+	// when
+	gotRsp, err = querier.VirtualStakingMaxCapLimits(sdk.WrapSDKContext(ctx), &types.QueryVirtualStakingMaxCapLimitsRequest{})
+	// then
+	require.NoError(t, err)
+	assert.Equal(t, 1, len(gotRsp.MaxCapInfos))
+	assert.Equal(t, myContract.String(), gotRsp.MaxCapInfos[0].Contract)
+	assert.Equal(t, myAmount, gotRsp.MaxCapInfos[0].Cap)
+
+	// set max cap for another contract
+	err = k.SetMaxCapLimit(ctx, sdk.AccAddress(rand.Bytes(32)), sdk.NewInt64Coin(sdk.DefaultBondDenom, 0))
+	require.NoError(t, err)
+
+	// when
+	gotRsp, err = querier.VirtualStakingMaxCapLimits(sdk.WrapSDKContext(ctx), &types.QueryVirtualStakingMaxCapLimitsRequest{})
+	// then
+	require.NoError(t, err)
+	assert.Equal(t, 2, len(gotRsp.MaxCapInfos))
```

Comment:
Please add an assert on the `gotRsp.MaxCapInfos` content, so that this is covered as well for regression.
You may find it difficult with the order as addresses are random. You could use `sdk.AccAddress(bytes.Repeat([]byte{n}, 32))` with n=1, 2, ... in this case for example or any pre-defined address set.